PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/AppServiceComboBox.java

```diff
@@ -59,9 +59,12 @@ public synchronized void refreshItemsWithDefaultValue(@NotNull T defaultValue) {
         this.addItem(defaultValue);
         subscription = this.loadItemsAsync()
                            .subscribe(items -> DefaultLoader.getIdeHelper().invokeLater(() -> {
-                               items.forEach(this::addItem);
-                               this.resetDefaultValue(defaultValue);
-                               this.setLoading(false);
+                               synchronized (AppServiceComboBox.this) {
+                                   AppServiceComboBox.this.removeAllItems();
+                                   items.forEach(this::addItem);
+                                   this.resetDefaultValue(defaultValue);
```

Comment:
is it enough if we just synchronized line 62 & 63 ？

Reply:
Will refactor the code in later pr.